lib/model/model.py

```diff
@@ -83,6 +83,23 @@ def respond(self, messages: Union[List[schemas.Message], schemas.Message]) -> Li
             message.body.result = self.get_response(message)
         return messages
 
+
+    @classmethod
+    def validate_input(cls, data: Dict) -> None:
+        """
+        Validate input data. Must be implemented by child classes.
+        """
+        raise NotImplementedError
```

Comment:
can this check for `id`, `callback_url`, and `parameters`?   can be invoked by subclasses that override validate_input?

Reply:
I guess we would have to make sure we call the super constructor every time we override it, otherwise it won't get enforced. but we already type check that those fields exist, and return error if they don't. those types of errors get checked centrally in `schemas.py` (through type definitions and `parse_input_message()`). but if you think still doing it would make sense I can be persuaded.

Reply:
> we already type check that those fields exist

then no need to check again here!